### DISCUSSION TITLE
Cycling through suggestions on macOS

### DISCUSSION THREAD

**Discussion by gwhobbs** [community/community#6762] [Copilot Conversations] [answered]
In the JetBrains plugin on macOS, the default keybindings for previous / next suggestion are `option-[` and `option-]`.

However, those key combinations are already assigned by the OS to open (slanted / curved) double quote and open (slanted / curved) single quote, respectively (see [this page](https://practicaltypography.com/straight-and-curly-quotes.html) for more info about different standard quote characters). Since that behavior is consistent across the OS and sometimes useful in the context of writing e.g. strings that will be displayed to a user, overriding the system behavior here seems likely undesirable.

`cmd-[` and `cmd-]` are already taken as well, but perhaps there is a better default, e.g. maybe `control-[` and `control-]`?

As an aside, the previous / next suggestion functionality seems to not be working currently in WebStorm, even after remapping to different keys, and even when selecting the options from the menu bar—but I am guessing this is a known problem with the beta JetBrains plugin.

**Comment by jansorg** [comment, marked as answer]
Thanks for raising this issue! We'll investigate if there's a better shortcut we could use on macOS.

Previous / next is currently only available for Java and Python files, where up to 3 completions are buffered.

**Reply by timmyimprint** [reply]
could we update the documentation to note this? the getting started guides says with JetBrains these languages are supported: "Python, JavaScript, TypeScript, Ruby, Java, and Go", so I was very confused when Go option-[ was failing!

**Reply by spacecat** [reply]
Yes, please update the docs asap. I almost spent half a day trying solving this :)

**Comment by daniel** [comment]
Also make the keyboard shortcuts configurable or use keyboard shortcuts that work on any keyboard (if that's possible?). "cmd-[" doesn't work on a Swedish/Nordic keyboard (I need to press Option+8 to make a "[".

**Reply by gwhobbs** [reply]
Hmm, since `cmd-[` is also a standard shortcut for decreasing indentation, I think that probably puts `cmd-[` out of the running. Maybe `control-[` then?

Edit: ah, I see now that the issue you are pointing to is a difficulty in typing the `[` character itself in some keyboard layouts. I believe the keymap is already configurable (at least in the JetBrains plugin). In fact, I've already changed the default keymapping to `control-[` on my system. 

I'm obviously biased as I have dedicated square bracket keys, but I think it's actually a good default for cycling through suggestions, since bracket keys are used for similar functions elsewhere in the OS like cycling through tabs in almost every app with tabs (including Finder, Safari, Chrome, Sublime, and so on—with cmd and shift modifiers). I'm curious if e.g. Safari and other apps have different default keybindings on Swedish/Nordic keyboards, or if those have to be manually set to something sensible?

**Reply by daniel** [reply]
I don't think Safari or any other Mac apps have different default keybindings for different keyboard layouts. I always remap next/previous tab in System Preferences - Keyboard - Shortcuts - App Shortcuts to Cmd+ä and Cmd+ö.

I found a way to remap the suggestions shortcuts in VsCode now, it was actually already documented [here](https://github.com/github/copilot-docs/blob/main/docs/visualstudiocode/configuring.md).
